content/docs/troubleshooting/webhook.md

```diff
@@ -1,56 +1,1073 @@
-description: |
-    Learn how to diagnose problems with the cert-manager webhook, such as connection errors or invalid or missing TLS certificate errors
+title: The Definitive Debugging Guide for the cert-manager Webhook Pod
+description: 'The cert-manager webhook is a Pod that runs as part of your cert-manager installation. When applying a manifest with kubectl, the Kubernetes API server calls the cert-manager webhook over TLS to validate your manifests. This guide helps you debug connectivity issues between the Kubernetes API server and the cert-manager webhook Pod.'
 ---
 
-Learn how to diagnose problems with the cert-manager webhook, such as connection errors or invalid or missing TLS certificate errors.
+The cert-manager webhook is a pod that runs as part of your cert-manager
+installation. When applying a manifest with `kubectl`, the Kubernetes API server
+calls the cert-manager webhook over TLS to validate your manifests. This guide
+helps you debug connectivity issues between the Kubernetes API server and the
+cert-manager webhook pod.
 
-## Overview
+The error messages listed in this page are encountered while installing or
+upgrading cert-manager, or shortly after installing or upgrading cert-manager
+when trying to create a Certificate, Issuer, or any other cert-manager custom
+resource.
 
-You may encounter errors when creating cert-manager resources indicating that the webhook is unreachable.
-In this case, it is advised to first check the [compatibility](../installation/compatibility.md) of your environment
-and take necessary actions outlined there.
-Then refer to the [known webhook problems](../concepts/webhook.md#known-problems-and-solutions).
-And finally, if you are still having problems follow the steps below.
+> Last updated on 26 August 2022 (cert-manager 1.9).
 
-## Check the webhook TLS certificates
+In the below diagram, we show the common pattern when debugging an issue with
+the cert-manager webhook: when creating a cert-manager custom resource, the API
+server connects over TLS to the cert-manager webhook pod. The red cross
+indicates that the API server fails talking to the webhook.
 
-The Kubernetes API server will load the CA content from the webhook configuration and use that to verify the serving certificate presented by the webhook server when the TLS connection is established.
+<img alt="Diagram that shows a kubectl command that aims to create an issuer resource, and an arrow towards the Kubernetes API server, and an arrow between the API server and the webhook that indicates that the API server tries to connect to the webhook. This last arrow is crossed in red." src="/images/troubleshooting/webhook-pod-networking-diagram.png" width="500"/>
 
-Get the webhook configuration and check the `caBundle` value.
-For example, to check the `ValidatingWebhookConfiguration`:
+The rest of this document presents the known error messages that you may
+encounter.
 
+## Error 1: `connect: connection refused`
```

Comment:
I kind of agree with you about it being a mistake to give these numbers.
Consider removing the numbers but keeping the prefix to only `Error: ...`

Reply:
I agree. I fixed it in aa48788e0d977b8f26dba3ec74362c305d9b5434.